pkg/metrics/metrics.go

```diff
@@ -0,0 +1,125 @@
+/*
+   Copyright 2014 CoreOS, Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Package metrics provides minimalist instrumentation for client side in
+// pull-based monitoring.
+package metrics
+
+import (
+	"bytes"
+	"expvar"
+	"fmt"
+	"sync"
+)
+
+// Counter is a number that increases over time monotonically.
+type Counter interface {
+	Add(delta int64)
+	String() string
+}
+
+// Gauge returns instantaneous value that is expected to fluctuate over time.
+type Gauge interface {
+	Set(value int64)
+	String() string
+}
+
+// Group aggregates counters, gauges and sub-groups.
+type Group struct {
+	// clear operation needs exclusive access to map
+	mu sync.RWMutex
```

Comment:
i don't want to export Lock and Unlock functions, so i set it as private member.
comment removed.